solr/core/src/java/org/apache/solr/servlet/DirectSolrConnection.java

```diff
@@ -114,14 +111,7 @@ public String request(SolrRequestHandler handler, SolrParams params, String body
       }
 
       // Now write it out
-      QueryResponseWriter responseWriter = core.getQueryResponseWriter(req);
-      if (responseWriter instanceof BinaryResponseWriter) {
-        return ((BinaryResponseWriter) responseWriter).serializeResponse(req, rsp);
-      } else {
-        StringWriter out = new StringWriter();
-        responseWriter.write(out, req, rsp);
-        return out.toString();
-      }
+      return req.getResponseWriter().writeToString(req, rsp);
```

Comment:
This pattern happens in many places.  The existence of it and repeating was a code-smell that the hierarchy was wrong; promoting my efforts here